doc/man3/OSSL_INDICATOR_set_callback.pod

```diff
@@ -0,0 +1,97 @@
+=pod
+
+=head1 NAME
+
+OSSL_INDICATOR_set_callback,
+OSSL_INDICATOR_get_callback - specify a callback for FIPS indicators
+
+=head1 SYNOPSIS
+
+ #include <openssl/indicator.h>
+
+ void OSSL_INDICATOR_set_callback(OSSL_LIB_CTX *libctx, OSSL_CALLBACK *cb,
+                                  void *cbarg);
+ void OSSL_INDICATOR_get_callback(OSSL_LIB_CTX *libctx, OSSL_CALLBACK **cb,
+                                  void **cbarg);
+
+=head1 DESCRIPTION
+
+OSSL_INDICATOR_set_callback() sets a user callback I<cb> associated with a 
+I<libctx> that will be called when a non approved FIPS operation is detected.
+The argument <cbarg> will be passed to the user callback.
+The users callback may be triggered multiple times during an algorithm operation
+to indicate different approved mode checks that have failed.
+
+Non approved operations may only occur if the user has deliberately chosen to do
+so (either by setting a global FIPS configuration option or via an option in an
+algorithms operation context).
+
+The user callback will receive the following L<OSSL_PARAM(3)> objects.
```

Comment:
Would arguments for these be better than params?  There is no preclusion about calls across the provider boundary having some typed arguments.

That is: 
```c
typedef int (*indicator_callback)(const char *type, const char *description, void *arg);
```

Params give us flexibility going forwards but we're never not going to have these two.  It might be worth considering a params argument in addition...

Reply:
I wanted to keep it completely generic with the ability to add additional params later if ever required.

Reply:
What about `typedef int (*indicator_callback)(const char *type, const char *description, OSSL_PARAM *params, void *arg);` then?  The type and description will also be there, so they make sense as arguments.